numpy/linalg/linalg.py

```diff
@@ -226,6 +227,22 @@ def _assertNoEmpty2d(*arrays):
         if _isEmpty2d(a):
             raise LinAlgError("Arrays cannot be empty")
 
+def transpose(a):
+    """
+    Transpose each matrix in a stack of matrices.
```

Comment:
TODO: expose this publicly?

Reply:
We need something like this, but I think it should be an array method as well. Maybe something like `a.M(atrix)T(ranspose`. When the `*.T` method was proposed there was discussion of what it should do, transposing matrices in a stack was one of ideas. I could also see a vector transpose, `a.VT` which just appends 1 to the dimensions. These ideas need discussion on the list.

Reply:
Should I add a leading underscore to the name for now?

Reply:
@eric-wieser That would help distinguish it from `np.transpose` and avoid some confusion. Long term, it probably needs a different name.

Reply:
At any rate, this isn't going to appear in the docs at the moment, so isn't really public API yet.

I'll send something around the mailing list requesting a name at some point. I think we can start off with the conservative `assert ndim >= 2` semantics initially, and put off the decision in #7495 and #9530